frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/OptionListEditor/LibraryOptionsEditor/LibraryOptionsEditor.tsx

```diff
@@ -0,0 +1,94 @@
+import type { Option } from 'app-shared/types/Option';
+import React, { createRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  StudioCodeListEditor,
+  StudioModal,
+  StudioAlert,
+  StudioParagraph,
+  StudioButton,
+} from '@studio/components';
+import type { CodeListEditorTexts } from '@studio/components';
+import { PencilIcon, TrashIcon } from '@studio/icons';
+import { usePreviewContext } from 'app-development/contexts/PreviewContext';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useUpdateOptionListMutation } from 'app-shared/hooks/mutations';
+import { useOptionListEditorTexts } from '../../../hooks';
+import type { OptionListEditorProps } from '../OptionListEditor';
+import classes from './LibraryOptionsEditor.module.css';
+
+type LibraryOptionsEditorProps = {
+  optionsList: Option[];
+  handleDelete: () => void;
+} & Pick<OptionListEditorProps, 'component'>;
+
+export function LibraryOptionsEditor({
+  component,
+  optionsList,
+  handleDelete,
+}: LibraryOptionsEditorProps): React.ReactNode {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { doReloadPreview } = usePreviewContext();
+  const { mutate: updateOptionList } = useUpdateOptionListMutation(org, app);
+  const editorTexts: CodeListEditorTexts = useOptionListEditorTexts();
+  const modalRef = createRef<HTMLDialogElement>();
+
+  const optionListHasChanged = (options: Option[]): boolean =>
+    JSON.stringify(options) !== JSON.stringify(optionsList);
+
+  const handleOptionsChange = (options: Option[]) => {
+    if (optionListHasChanged(options)) {
+      updateOptionList({ optionListId: component.optionsId, optionsList: options });
+      doReloadPreview();
+    }
+  };
+
+  const codeListLabels: string = optionsList
+    .map((option: Option) => `${option.label || t('general.empty_string')}`)
+    .join(' | ');
+
+  return (
+    <div className={classes.container}>
+      <StudioParagraph className={classes.label}>{component.optionsId}</StudioParagraph>
+      <StudioParagraph size='sm' className={classes.codeListLabels} variant='short'>
+        {codeListLabels}
+      </StudioParagraph>
```

Comment:
These components can also be moved to a separate component so we do not need to duplicate the implementation and css.